agent/src/main.rs

```diff
@@ -171,14 +170,34 @@ async fn main() -> Result<()> {
         );
     }
 
+    let mut roles: Vec<ServalRole> = Vec::new();
+    if blob_path.is_some() {
+        log::info!("serval agent blob store mounted; path={blob_path:?}");
+        roles.push(ServalRole::Storage);
+    }
     if should_run_jobs {
-        // todo: actually start polling job queue for work to do
         log::info!("job running enabled");
-        advertise_service("serval_runner", port, &instance_id, None)?;
+        roles.push(ServalRole::Runner);
     } else {
         log::info!("job running not enabled (or not supported)");
     }
 
+    let mesh_port: u16 = match std::env::var("MESH_PORT") {
+        Ok(port_str) => port_str.parse::<u16>().unwrap_or(8181),
+        Err(_) => 8181,
+    };
+
+    let metadata = PeerMetadata::new(
+        Uuid::new_v4().to_string(),
+        format!("http://{}:{}", host, port),
+        roles,
```

Comment:
It probably doesn’t actually matter, and we certainly don't have to change anything right now, but should we encode the roles in the identity payload as bit masks rather than as strings? The identity payload is included in every single kaboodle packet, so little optimizations add up.

Surely every conceivable role we'd have would fit into a single byte, and since we have versioning (brilliant idea, by the way—I would not have thought of that) we can expand it if we ever need more than 8 roles.

If you agree, I can make a GitHub issue so we don't lose the idea.